### ISSUE TITLE
Using AppListener

### ISSUE THREAD

**Issue by yigitlevent** [ultralight-ux/Ultralight#145]
So, I have a (possibly dumb) question. When I try the LoadListener to work with the view, it just works without much modification. But I cannot get the AppListener to work. 

I just want to test out few features so I really didn't want to bother with a custom Driver and all that. Do I need to write my own Run() function, drivers and all that as it was stated in the README file, or am I doing something wrong? 

```
class AppL : public ultralight::AppListener {
public:
	AppL() {}
	virtual ~AppL() {}
	virtual void OnUpdate(ultralight::View* caller) { std::cout << "This doesn't work" << std::endl; }
};

class LoadL : public ultralight::LoadListener {
public:
	LoadL() {}
	virtual ~LoadL() {}
	virtual void OnBeginLoading(ultralight::View* caller) {}
	virtual void OnFinishLoading(ultralight::View* caller) {}
	virtual void OnUpdateHistory(ultralight::View* caller) {}
	virtual void OnDOMReady(ultralight::View* caller) { std::cout << "This works" << std::endl;  }
};

int main(int, char**) {
	auto application_ = ultralight::App::Create();
	auto window_ = ultralight::Window::Create(application_->main_monitor(), 300, 300, false, ultralight::kWindowFlags_Titled);

	window_->SetTitle("Basic App Test");
	application_->set_window(window_);

	auto overlay_ = ultralight::Overlay::Create(window_, window_->width(), window_->height(), 0, 0);

	overlay_->view()->set_load_listener(new LoadL()); // Works
	overlay_->view()->LoadHTML("<center>Hello World!</center>");

	application_->set_listener(new AppL()); // Doesn't work

	application_->Run();

	return 0;
}
```

**Comment by adamjs** [comment]
There's a typo in your `OnUpdate()` function, it doesn't take any params (make sure to use the `override` keyword so that your compiler can warn you about these issues :))

The `ultralight-quick-start` repo demonstrates how to use AppListener: <https://github.com/ultralight-ux/ultralight-quick-start/blob/master/src/MyApp.cpp#L67>

**Comment by aosenkidu** [comment]
On 28 Jul 2023 04:07, Adam Simmons ***@***.***> wrote:
Closed #145 as completed.
